Assets/LeapMotion/Core/Resources/HandData.cginc

```diff
@@ -0,0 +1,311 @@
+﻿// HandData.cginc
+
+// ==============
+// Fingertip Data
+// ==============
+
+uniform float4 _Leap_LH_Fingertips[5];
+uniform float4 _Leap_RH_Fingertips[5];
+
+uniform float4 _Leap_LH_PalmTriangles[15];
+uniform float4 _Leap_RH_PalmTriangles[15];
+
+uniform float4 _Leap_LH_FingerSegments[28];
+uniform float4 _Leap_RH_FingerSegments[28];
+
+// =========
+// Functions
+// =========
+
+#define IDENTITY_MATRIX4x4 { 1.0, 0.0, 0.0, 0.0, \
+                             0.0, 1.0, 0.0, 0.0, \
+                             0.0, 0.0, 1.0, 0.0, \
+                             0.0, 0.0, 0.0, 1.0 }
+
+float4x4 Leap_HandData_Preprocess_Matrix = IDENTITY_MATRIX4x4;
+
+void Leap_ClearPreprocessMatrix() {
+  //Leap_HandData_Preprocess_Matrix = IDENTITY_MATRIX4x4;
+}
+
+//----------------------
+// Distance Support (iq)
+//----------------------
+// http://www.iquilezles.org/www/articles/distfunctions/distfunctions.htm
+
+float Leap_SqrLength(float3 v) {
+  return v.x * v.x + v.y * v.y + v.z * v.z;
+}
+
+float Leap_SqrDistLine(float3 p, float3 a, float3 b) {
+  float3 pa = p - a, ba = b - a;
+  float h = clamp( dot(pa,ba)/dot(ba,ba), 0.0, 1.0 );
+  return Leap_SqrLength( pa - ba*h );
+}
+
+float Leap_SqrDistLine_WithThickness(float3 p, float3 a, float3 b, float approxR) {
+  return max(Leap_SqrDistLine(p, a, b) - approxR * approxR, 0);
+}
+
+float dot2(float3 v) { return dot(v,v); }
```

Comment:
dot2 is the same as Leap_SqrLength, we should probably keep one and ditch the other.

Reply:
yup yup